src/main/scala/bio/ferlab/ferload/services/AuthorizationService.scala

```diff
@@ -93,6 +113,55 @@ class AuthorizationService(authConfig: AuthConfig, backend: SttpBackend[IO, Fs2S
 
   }
 
+  /**
+   * Fetch only resources user has access from a provided input list of resources ids. Return a list resources id the user is authorized. Otherwise, return an error.
+   *
+   * @param token     the token to validate
+   * @param resources the resources to access
+   * @return ta list resources id the user is authorized. Otherwise, return errors (Unauthorized, Forbidden, NotFound).
+   */
+  def authLogicAuthorizationForUser(token: String, resources: Seq[String]): IO[Either[(StatusCode, ErrorResponse), User]] = {
+    val r: IO[User] = for {
+      partyToken <- requestPartyToken(token, resources)
+      permissionToken <- introspectPartyToken(partyToken)
+    } yield {
+      val value: Set[Permissions] = permissionToken.permissions.map(_.toSet).getOrElse(Set.empty)
+      User(partyToken, value)
+    }
+
+    r.map {
+        case User(_, permissions) => Right(User(token, permissions))
+      }
+      .recover {
+        case HttpError(_, statusCode) if Seq(StatusCode.Unauthorized, StatusCode.Forbidden).contains(statusCode) => Left((statusCode, ErrorResponse("Unauthorized", statusCode.code))).withRight[User]
+        case e: HttpError[String] if e.statusCode == StatusCode.BadRequest && e.body.contains("invalid_resource") => Left((StatusCode.NotFound, ErrorResponse("Not Found", 404))).withRight[User]
+      }
+
+  }
+
+  /**
+   * Fetch all resources user has access. Return a list resources id the user is authorized. Otherwise, return an error.
+   *
+   * @param token     the token to validate
+   * @return ta list resources id the user is authorized. Otherwise, return errors (Unauthorized, Forbidden, NotFound).
+   */
+  def authLogicAuthorizationForUser(token: String): IO[Either[(StatusCode, ErrorResponse), Seq[String]]] = {
+    val r: IO[Seq[String]] = for {
+      userPermissions <- requestUserPermissions(token)
+    } yield {
+      userPermissions.map(_.rsid)
+    }
+
+    r.map {
+        case resource: Seq[String] => Right(resource)
+      }
+      .recover {
+        case HttpError(_, statusCode) if Seq(StatusCode.Unauthorized, StatusCode.Forbidden).contains(statusCode) => Left((statusCode, ErrorResponse("Unauthorized", statusCode.code))).withRight[Seq[String]]
+        case e: HttpError[String] if e.statusCode == StatusCode.BadRequest && e.body.contains("invalid_resource") => Left((StatusCode.NotFound, ErrorResponse("Not Found", 404))).withRight[Seq[String]]
```

Comment:
Même question

Reply:
Best to leave as is. Less confusing this response.